### PR TITLE
Application class

### PR DESCRIPTION
**Pull request by JarrettSJohnson**
Pulling much of the SDL boiler plate into a class and letting RAII handle it.

### CODE REVIEW

File: src/Application.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	Application::Application(){
+		/* Initialize SDL */
+		if (SDL_Init(SDL_INIT_EVERYTHING) != 0)
+		{
+			std::cerr << "SDL_Init Error: " << SDL_GetError() << std::endl;
+			m_quit = true;
+		}
+
+		/* Create window */
+		m_window = SDL_CreateWindow("Game", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, 1280, 720, SDL_WINDOW_RESIZABLE);
```
Comment (Itay2805) [inline src/Application.cpp:16]: should we attempt to create a window if we failed to init SDL?

File: src/Application.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		if (m_window == nullptr)
+		{
+			std::cerr << "SDL_CreateWindow Error: " << SDL_GetError() << std::endl;
+			m_quit = true;
+		}
+
+		/* Create renderer */
+		m_renderer = SDL_CreateRenderer(m_window, -1, SDL_RENDERER_ACCELERATED);
+		if (m_renderer == nullptr)
```
Comment (Itay2805) [inline src/Application.cpp:26]: same as above

File: src/Application.cpp (hunk 1 above)
Comment (JarrettSJohnson) [reply src/Application.cpp:16]: Good question. The only other idea I had was to throw an exception if this failed, but I'm going to just assume no exceptions are wanted in this project?

Comment (JarrettSJohnson) [reply src/Application.cpp:16]: Alternatively we can set m_quit=true and return from the constructor early (which would've been a much better approach than what I submitted)

Comment (Itay2805) [reply src/Application.cpp:16]: yeah I think the second option would be more ideal. but both are valid options, it is a question if we want exceptions for error handling or not.

**Comment by JarrettSJohnson** [comment]
pushed new commit with early returns.

**Review by Itay2805 — APPROVED** [review]
Nice work :)

**Pull request merged**